### PR TITLE
Add Foreach Function

### PR DESCRIPTION
**Pull request by kwrobert**
Added a F.foreach function similar to the F.extend and F.update functions. 

If there is a key in the configuration that matches "F.foreach", a special function is called that modifies the structure of the dictionary via a parameterized for loop. It was a pretty simple addition, so looking at the diff should make it clear what this new function does. I added it because this library is really nice, but I needed this function to cover 100% of my use cases (for now). 

I created a separate branch for these modifications, so we can discuss any changes you'd like me to make (in the event you even want to merge this) without the history on my master branch getting all cluttered. Please let me know how you like me to merge/rebase onto my master branch. 

I'd be happy to continue contributing! I'll definitely continue to use this in one of my projects, so I have a vested interest in ongoing maintenance.

### CODE REVIEW

**Comment by kororo** [comment]
Hi @kwrobert,

I been thinking about this feature in my wish-list. Thanks for the contribution, I would love to hear lots of your use cases and opinion on the library. I like your add-on,  I will merged this on into master for now, once we have lots of feature, I will put into pypi.

Meanwhile, from this valuable insight, I will put into features to-do:
- Feature to display warning and error messages (for example, missing parameters, logging purposes)
- Add options to extend structure in template (for example, https://github.com/defunkt/pystache), the idea is render the values in string, convert it to object and attach it back. I think this mostly cover all of your current use case here. This will also inherit lots of feature such as loop, decision blocks
- Separate current functions into built-in and restructure it (I think this will be the first break-changes to organise all the built-in functions into more better way)
- Add code conduct, issue template into git project.

In summary on your commit, I am happy with your contribution, it has that the project needs:
- new feature add-on
- well tested (100% coverage)
- has error handling

Just missing little, could you please to add info into README.rst for the new feature? 

PS. If you like to contribute more, I can add you into contributors.

Thanks!!
